code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -150,7 +187,37 @@ public Account createAccount(String name, String kind, String externalId, String
         if (StringUtils.isNotEmpty(externalType)) {
             properties.put(ACCOUNT.EXTERNAL_ID_TYPE, externalType);
         }
-        return resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class, properties));
+        return resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class,
+                properties));
+    }
+
+    @Override
+    public Credential createUserPassword(Account account, String username, String password) {
+        Map<Object, Object> properties = new HashMap<>();
+        if (StringUtils.isBlank(username) || StringUtils.isBlank(password)) {
+            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "UserCredentialCreation", "Cannot have blank username or password.", null);
+        }
+        properties.put(CREDENTIAL.PUBLIC_VALUE, username);
+        properties.put(CREDENTIAL.SECRET_VALUE, password);
+        properties.put(CREDENTIAL.KIND, LocalAuthConstants.PASSWORD);
+        properties.put(CREDENTIAL.ACCOUNT_ID, account.getId());
+        return resourceDao.createAndSchedule(Credential.class, objectManager.convertToPropertiesFor(Credential.class, properties));
+    }
+
+    @Override
+    public Identity getIdentity(Long id) {
+        Account account = getAccountById(id);
+        if (account == null || account.getKind().equalsIgnoreCase(ProjectConstants.TYPE)) {
+            return null;
+        }
+        Credential credential = create()
+                .selectFrom(CREDENTIAL)
+                .where(CREDENTIAL.KIND.equalIgnoreCase("password")
+                        .and(CREDENTIAL.ACCOUNT_ID.eq(id))
+                .and(CREDENTIAL.STATE.equalIgnoreCase(CommonStatesConstants.ACTIVE))).fetchAny();
```

Comment:
I would say not allow multiple `password` credentials for a user in the API.  That will have to be enforced in the API though, simple db uniqueness won't work.  Then when you're reading them back, if you get more than one row (which should be impossible), bail out.

I know this differs from what we wrote out yesterday, and probably not easy to implement in the framework.  But for discussion, as an alternative to `POST /v1/passwords` to create and `GET /v1/passwords?accountId=:id` + `POST /v1/passwords/:id?action=changeSecret` to create and change passwords, we could have a single `/v1/accounts/:id?action=setPassword {publicValue: 'me', secretValue: 'newPass', [oldSecretValue: 'oldPass']}` action that creates-or-updates the `password` resource.  Then the API consumer doesn't have to know about the separate resource at all and it's clear that there is a single user/password for that account.